x/attribute/types/msgs.go

```diff
@@ -36,7 +36,10 @@ func (msg MsgAddAttributeRequest) ValidateBasic() error {
 	if _, err := sdk.AccAddressFromBech32(msg.Owner); err != nil {
 		return err
 	}
-	a := NewAttribute(msg.Name, msg.Account, msg.AttributeType, msg.Value, msg.ExpirationDate)
+	if len(msg.ConcreteType) > 200 {
+		return fmt.Errorf("concrete_type exceeds maximum length of 200 characters")
+	}
+	a := NewAttribute(msg.Name, msg.Account, msg.AttributeType, msg.Value, msg.ExpirationDate, msg.ConcreteType)
 	return a.ValidateBasic()
```

Comment:
Since you're calling `a.ValidateBasic()` here, and that does the length check on the concrete type, you don't also need to validate it here. Same with below.

It's even more crucial to remove these duplicate validation pieces because that 200 is hard coded in each instead of being a variable.